### ISSUE TITLE
gloo-net: Cannot compile with `http` feature but not `json` feature

### ISSUE THREAD

**Issue by mpfaff** [rustwasm/gloo#227]
### Describe the Bug

The `.json` helper on `Request` (added in #204) is not conditional on the `json` feature. The same goes for the `serde::Serialize` import at the top of the file.

### Steps to Reproduce

1. Add `gloo-net = { version = "0.2.1", default-features = false, features = [ "http" ] }` to Cargo.toml
2. Run `cargo build`

### Expected Behavior

The `.json` helper is not available because the `json` feature is not enabled and the crate compiles.

### Actual Behavior

The crate fails to compile because of the unresolved import:

```
error[E0432]: unresolved import `serde`
  --> ~/.cargo/registry/src/github.com-1ecc6299db9ec823/gloo-net-0.2.1/src/http/mod.rs:21:5
   |
21 | use serde::Serialize;
   |     ^^^^^ use of undeclared crate or module `serde`

error[E0433]: failed to resolve: use of undeclared crate or module `serde_json`
   --> ~/.cargo/registry/src/github.com-1ecc6299db9ec823/gloo-net-0.2.1/src/http/mod.rs:176:20
    |
176 |         let json = serde_json::to_string(value)?;
    |                    ^^^^^^^^^^ use of undeclared crate or module `serde_json`
```

**Comment by hamza1311** [comment]
Fixed in gloo-net v0.2.2

Thanks for the report!

**Comment by mpfaff** [comment]
Thank you for the quick fix!
